Content.Client/Stunnable/StunSystem.cs

```diff
@@ -1,9 +1,227 @@
+﻿using System.Numerics;
+using Content.Shared.Mobs;
 using Content.Shared.Stunnable;
+using Robust.Client.Animations;
+using Robust.Client.GameObjects;
+using Robust.Shared.Animations;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+using Robust.Shared.Utility;
 
 namespace Content.Client.Stunnable
 {
     public sealed class StunSystem : SharedStunSystem
     {
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly IGameTiming _timing = default!;
+        [Dependency] private readonly SpriteSystem _spriteSystem = default!;
 
+        private readonly int[] _sign = [-1, 1];
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<StunnedComponent, ComponentInit>(OnComponentInit);
+            SubscribeLocalEvent<StunnedComponent, MobStateChangedEvent>(OnMobStateChanged);
+            SubscribeLocalEvent<StunnedComponent, AppearanceChangeEvent>(OnAppearanceChanged);
+        }
+
+        /// <summary>
+        ///     Add stun visual layers
+        /// </summary>
+        private void OnComponentInit(Entity<StunnedComponent> entity, ref ComponentInit args)
+        {
+            if (_timing.IsFirstTimePredicted)
+                return;
+
+            if (!TryComp<SpriteComponent>(entity, out var sprite))
+                return;
+
+            _spriteSystem.LayerMapReserve((entity, sprite), StunVisualLayers.StamCrit);
+            _spriteSystem.LayerSetVisible((entity, sprite), StunVisualLayers.StamCrit, false);
+            _spriteSystem.LayerSetOffset((entity, sprite), StunVisualLayers.StamCrit, new Vector2(0, 0.3125f));
+
+            _spriteSystem.LayerSetRsi((entity.Owner, sprite), StunVisualLayers.StamCrit, new ResPath("Mobs/Effects/stunned.rsi"));
+
+            UpdateAppearance((entity, sprite, entity.Comp));
+        }
+
+        private void OnAppearanceChanged(Entity<StunnedComponent> entity, ref AppearanceChangeEvent args)
+        {
+            if (args.Sprite != null)
+                UpdateAppearance((entity, args.Sprite, entity.Comp));
+        }
+
+        private void UpdateAppearance(Entity<SpriteComponent?, StunnedComponent> entity)
```

Comment:
```suggestion
        private void UpdateAppearance(Entity<StunnedComponent, SpriteComponent?> entity)
```
keep it consistent with the other methods